nselib/rdp.lua

```diff
@@ -113,23 +134,58 @@ Packet = {
     end,
 
     parse = function(data)
+
+      local tag_decoder = {}
+
+      tag_decoder["\x0A"] = function( self, encStr, elen, pos )
+        return self.decodeInt(encStr, elen, pos)
+      end
+
       local ccr = Packet.ConfCreateResponse:new()
-      local pos = 67
 
-      while data:len() > pos do
-        block_type, block_len  = string.unpack("<I2I2", data, pos)
+      local pos = 3
+
+      local decoder = asn1.ASN1Decoder:new()
+      decoder:registerTagDecoders( tag_decoder )
+
+      local response_result, userdata
+      _, pos = decoder.decodeLength(data, pos)
```

Comment:
`_` needs to be local here. You could do `local _, pos = decoder.decodeLength(data, 3)` and eliminate the extra declaration on line 146.